tests/versions/index.mjs

```diff
@@ -102,6 +104,50 @@ let hasError = false;
   }
 }
 
+{
+  const data = await fetch('https://registry.npmjs.org/search-insights')
+    .then((res) => res.json())
+    .catch(() => null);
+
+  if (
+    !data ||
+    !data.versions ||
+    !data['dist-tags'] ||
+    !data['dist-tags'].latest
+  ) {
+    console.error('Unable to fetch versions from npm');
+    console.error("Can't check if search-insights is up to date");
+  } else {
+    const { latest } = data['dist-tags'];
+    const dependency =
+      require('../../packages/instantsearch.js/package.json').dependencies[
+        'search-insights'
+      ].replace('^', '');
+    const [, middleware] = fs
+      .readFileSync(
+        path.join(
+          dirname,
+          '../../packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts'
+        ),
+        'utf-8'
+      )
+      .match(/ALGOLIA_INSIGHTS_VERSION = '(.*)';/);
+
+    if (latest !== dependency || latest !== middleware) {
+      console.error(`search-insights is not up to date.`);
+      hasError = true;
```

Comment:
I'm not comfortable with failing our pipelines because we don't have the latest version of the Insights client. For example, if the latest is a new major, it might require more work to update on our end, which we might not be able to allocate right away.

An alternative, in addition to a "dependabot-like" scheduled PR, could be to still pass the CI but publish a comment on the PR (or add a check like Bundlesize with a warning).

Reply:
That's fair, I'll look into doing this in a PR instead